### ISSUE TITLE
Not all jump operand variants parsed properly.

### ISSUE THREAD

**Issue by ThatOneDamien** [naclsn/tree-sitter-nasm#4]
# Incorrect JMP instruction parsing

Sorry let me preface this by saying this is the first time I've ever tried contributing to a project in any form or fashion, so I am truly sorry if my ignorance causes any confusion or issues. Feel free to point out anything stupid I say/do.

Now for the issue I am facing: at the moment when an instruction is parsed, there are limited options for the operand formats, those being effective addresses, registers, and other forms of expressions. The issue arises when a valid jmp instruction is encountered that does not require the brackets for an effective address, but one of its operands is in the segment:offset format.

An example: 
```asm
jmp LabelA ; Regular jmp
jmp 0x0FF:0xF ; jmp that is parsed incorrectly
```

I understand the reason for this is very obvious, at the moment operands in the form segment:offset with no surrounding brackets are not being processed, but it results in some unwanted behavior in a valid nasm program (See picture below).

![Screenshot_20241116_172849](https://github.com/user-attachments/assets/6f278489-5f47-4829-af57-d1867cb08a16)

I have come up with a change that fixes the issue, but I am not sure of its robustness. I would love to submit a pull request for the proposed fix, but I also have never done that before and I don't want to format it improperly or something.

### Platform
Arch Linux 

### Terminal Emulator
Wezterm

### Editor
Neovim

**Comment by naclsn** [comment]
Sorry for the delayed response; I'm surprisingly busy lately.

Your message is very clean and with more self-confidence I think it could even be made directly into a PR!
This maybe doesn't apply everywhere (I haven't opened a lot of PR myself), but it's never really wrong to open one. The maintainer(s) always get a chance to review and give feedback, or (hopefully politely) dismiss it.

For the bug itself, well I simply didn't even know it was an existing syntax :/ thank you for pointing it out!

Not sure of the robustness of your fix? To be frank I'm not sure myself how this thing holds itself together "this well".
Testing is setup, with `npm run test`; it runs the parser against the files in `test/corpus/`. Rule of thumb to make a PR I suppose is it should pass testing (tho here I would be open to negotiate on edge-cases).
If your changes don't already include a test, then we'll see about adding one so it doesn't break again when fixing another thing later on. (Some of this snippet you provided would be enough; maybe in `source_line.txt`, doesn't matter that much.)

Lastly however, I don't know how nvim manages the grammar versions it uses; if there are additional steps to take so your fix gets used then you might have to do it yourself (for example, I know helix has a `language.toml` in the main repo that has to be updated each time...)

**Comment by ThatOneDamien** [comment]
I really appreciate the reassuring response, I just didn't want to make a fool of myself lol. Also you're all good, your response was very quick! 

One thing I wanted to note that I forgot to in the original post is that the instruction that causes the issue is quite niche, as that jmp instruction is only valid in 16 bit contexts from what I understand. I encountered it while experimenting with building an operating system.

Anyways, I will work on the changes and submit a pull request later that passes the testcases. I also just wanted to ask, when you clone the repo and run the test cases, is one of the macro testcases failing? After freshly cloning on my machine, building, and running, the header guard testcase fails. I actually also found a fix for that issue as well, but would it be best to possibly open a separate pull request?
